examples/flask/test_rock/app.py

```diff
@@ -73,6 +74,35 @@ def get_redis_database() -> redis.Redis | None:
     return g.redis_db
 
 
+def get_rabbitmq_connection():
+    """Get rabbitmq connection."""
+    if "rabbitmq" not in g:
+        if "RABBITMQ_HOSTNAME" in os.environ:
+            username = os.environ["RABBITMQ_USERNAME"]
+            password = os.environ["RABBITMQ_PASSWORD"]
+            hostname = os.environ["RABBITMQ_HOSTNAME"]
+            vhost = os.environ["RABBITMQ_VHOST"]
+            port = os.environ["RABBITMQ_PORT"]
+            credentials = pika.PlainCredentials(username, password)
+            parameters = pika.ConnectionParameters(hostname, port, vhost, credentials)
+            g.rabbitmq = pika.BlockingConnection(parameters)
```

Comment:
It seems easier to read to me if you directly return `pika.BlockingConnection(parameters)`

Reply:
The connection is stored in the flask application context [g](https://flask.palletsprojects.com/en/2.3.x/api/#flask.g). I did that mainly for consistency with the previous functions.

Reply:
Why returning it then? It seems unnecessary. I think I'd remove the return and change the function name so that it reflects the configuration change

Reply:
I am not sure what you mean. Do you suggest using something like? If that's the case I am not sure that is cleaner.

```
@app.route("/rabbitmq/receive")
def rabbitmq_receive():
    """Receive a message from "charm" queue in blocking form."""
    prepare_rabbitmq_connection_from_uri()
    if connection := g.rabbitmq_from_uri:
       ...
```

Reply:
if you agree I will let it this way for consistency. WDYT?